aten/src/ATen/CPUApplyUtils.h

```diff
@@ -538,18 +556,33 @@ inline void CPU_tensor_parallel_apply2(
   if (!_apply_preamble({tensor1, tensor2}))
     return;
   if (tensor1.ndimension() < 8 && tensor2.ndimension() < 8) {
-    parallel_for(
-        0,
-        tensor1.numel(),
-        grain_size,
-        [&tensor1, &tensor2, &op](int64_t begin, int64_t end) {
-          apply_op(
-              end - begin,
-              begin,
-              op,
-              strided_tensor_iter_fixed<scalar1, 8>(tensor1),
-              strided_tensor_iter_fixed<scalar2, 8>(tensor2));
-        });
+    if (tensor1.is_contiguous()) {
+      parallel_for(
+          0,
+          tensor1.numel(),
+          grain_size,
+          [&tensor1, &tensor2, &op](int64_t begin, int64_t end) {
+            apply_op(
+                end - begin,
+                begin,
+                op,
+                strided_tensor_iter_fixed<scalar1, 8, false>(tensor1),
```

Comment:
Looks like you haven't checked that `tensor2` is contiguous?

Reply:
Purposely as this apply operation primarily used in copy (clone) function where destination is contiguous and source not (as when they are both contiguous we do memcpy).

Reply:
Does that mean that `t1.copy_(t2)`, for t1 non-contiguous (or t2, didn't check the rest of the code) will give wrong results?

Reply:
It is only mean that it will use older non-optimized kernel. I can do kernel per each couple: (non-contiguous, non-contiguous).(contiguous,non-contiguous),... but it blows up exponentially for cpu_apply that takes 3 or more tensors.